legion/operator/pkg/controller/modeltraining/tectoncd.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package modeltraining
+
+import (
+	legionv1alpha1 "github.com/legion-platform/legion/legion/operator/pkg/apis/legion/v1alpha1"
+	"github.com/legion-platform/legion/legion/operator/pkg/apis/training"
+	operator_conf "github.com/legion-platform/legion/legion/operator/pkg/config/operator"
+	training_conf "github.com/legion-platform/legion/legion/operator/pkg/config/training"
+	"github.com/legion-platform/legion/legion/operator/pkg/legion"
+	"github.com/legion-platform/legion/legion/operator/pkg/repository/util/kubernetes"
+	"github.com/spf13/viper"
+	tektonv1alpha1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"path"
+)
+
+const (
+	pathToPackagerBin = "/opt/legion/trainer"
+	workspacePath     = "/workspace"
+	outputDir         = "output"
+)
+
+func generateTrainerTaskSpec(
+	trainingCR *legionv1alpha1.ModelTraining,
+	toolchainIntegration *training.ToolchainIntegration,
+) (*tektonv1alpha1.TaskSpec, error) {
+	mainPackagerStep, err := createMainPackagerStep(trainingCR, toolchainIntegration)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tektonv1alpha1.TaskSpec{
+		Steps: []tektonv1alpha1.Step{
+			createInitPackagerStep(trainingCR.Name),
```

Comment:
Maybe we should rename these func names to blabla**Trainer**Step ?

Reply:
Fixed